### ISSUE TITLE
DLL 'rJava' not found when installing 'ReporteRsjars'

### ISSUE THREAD

**Issue by steveb-cirina** [davidgohel/ReporteRs#152]
In short, I am having a problem installing `ReporteRs` / `ReporteRsjars`.

I am getting the following error on Windows 7 (I am a Mac and Linux person primarily, so I may be missing something obvious):

> DLL 'rJava' not found when installing 'ReporteRsjars'

when I attempt to install `ReporteRsjars`,  which gets installed as a dependency for `ReporteRs`.

Unlike others who have had the error when running `library(rJava)`, the `library` command works for me w/o a problem.  I tried a few `rJava::*` commands as well and they seem to work (I am new to `rJava`).

I posted this on Stack Overflow and here is the link with details:

[Error installing 'ReporteRs' and 'ReporteRsjars'](http://stackoverflow.com/questions/40644971/error-installing-reporters-and-reportersjars)

I have been trying a number of things but have not been able to resolve this. This includes modifying the path variable, setting and unsetting JAVA_HOME.

Any insights would be greatly appreciated.

-Steve

**Comment by davidgohel** [comment]
Hi

Oups, the `rJava` hell...

From what I can see, the issue is with 32bits. I think you are in the case where java is installed only in its 64bits version (the 32bits is not installed). Could you try the following?

```
install.packages("ReporteRs", contriburl=cran_dir_url, repos = cran_dir_url, 
  type = 'source', INSTALL_opts = "--no-multiarch")
```

It should build ReporteRs only for the 64 bits architecture.

**Comment by steveb-cirina** [comment]
Thanks for your reply and yes, this has been "`java` hell".  It is too bad the errors aren't something along the lines of "expecting both 32 & 64 bit java to be available ...".  The '--no-multarch' option must be handled automatically when installing a

I ended up installing on an "identical" machine (they are both the same hardware and OS) that has internet access and copying the R library directories to the machine that doesn't have these.

**Comment by steveb-cirina** [comment]
Thanks for your reply and yes, this has been "`java` hell".  It is too bad the errors aren't something along the lines of "expecting both 32 & 64 bit java to be available ...".  The '--no-multarch' option must be handled automatically when installing using CRAN (e.g. `install.packages('ReporteRs')`).

I ended up installing on an "identical" machine (they are both the same hardware and OS) that has internet access and copying the R library directories to the machine that doesn't have these.  Other than requiring a 2nd machine that is the "same", can this cause issues ?  The good side is there is no need to grab all of CRAN (or a targeted subset) and installs are quicker.  The down side is it requires a 2nd machine that is similar.

I will try your suggestion on the next install to see if it works.

**Comment by steveb-cirina** [comment]
The suggested fix (using `--no-multiarch`) worked!  I removed the library directories (`rJava`, `ReporteRsjars` and `ReporteRs`) and re-did the install using the local CRAN copy and the `--no-multiarch` option.

**Comment by davidgohel** [comment]
Thanks, I will then add this on SO also.

**Comment by steveb-cirina** [comment]
In the end, I guess I didn't know how to interpret the error message.

Thanks again, your help is appreciated.


Cheers,
-Steve

________________________________

**Comment by steveb-cirina** [comment]
If you put up your answer to my question, I can accept it.

W.r.t. SO etiquette, should I delete my answer if you put yours up ?

-Steve


________________________________
From: steven banville
Sent: Friday, November 18, 2016 11:37 AM
To: davidgohel/ReporteRs
Subject: RE: [davidgohel/ReporteRs] DLL 'rJava' not found when installing 'ReporteRsjars' (#152)


In the end, I guess I didn't know how to interpret the error message.

Thanks again, your help is appreciated.


Cheers,
-Steve

________________________________

**Comment by davidgohel** [comment]
No need, the answer is written so it's ok :) 
I upvoted your answer.

**Comment by steveb-cirina** [comment]
Thanks again for your help and for the up vote :-)

**Comment by IL2liu** [comment]
hello, I am having issues in installing "ReporteRs" package, and the code and error is as following:
>install.packages("ReporteRs")
Warning in install.packages :
  package ‘ReporteRs’ is not available (for R version 3.5.1)
R version 3.5.1 (2018-07-02)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 7 x64 (build 7601) Service Pack 1
Matrix products: default
#### 
I have tried many other ways, but it just doesn't work. 
If possible, could you please offer me some suggestions?
